### DISCUSSION TITLE
bug: System sleep prevention due to wake lock in Open-WebUI

### DISCUSSION THREAD

**Discussion by qdrop17** [open-webui/open-webui#4296] [General]
# Bug Report

## Description

**Bug Summary:**
OpenWebUI creates a wake lock in Chromium-based browsers that prevents the system from entering sleep mode automatically, even when the application is idle.

**Steps to Reproduce:**

1. Open OpenWebUI in a Chromium-based browser (e.g., Chrome, Brave).
2. Leave the application open but idle.
3. Observe that the system does not enter sleep mode automatically as it normally would.
4. Close OpenWebUI.
5. Observe that the system can now enter sleep mode normally.

In Fedora / Gnome, this can be accelerated a bit by executing this command:

- With Open WebUI open:

```
~$ gnome-session-inhibit --list
/opt/brave.com/brave/brave: Blink Wake Lock (suspend, idle)
mutter: idle-inhibit (idle)
```

- With Open WebUI closed:

```
~$ gnome-session-inhibit --list
No inhibitors
```

**Expected Behavior:**
The system should be able to enter sleep mode automatically based on the user's system settings, regardless of whether OpenWebUI is open.

**Actual Behavior:**
When OpenWebUI is open, a wake lock is created that prevents the system from entering sleep mode automatically. This wake lock is removed immediately upon closing OpenWebUI.

## Environment

- **Open WebUI Version**: v0.3.10
- **Operating System**: Fedora 40 with Gnome; up to date running Wayland
- **Browser**: Brave 1.68.134 Chromium: 127.0.6533.88 (Official Build) (64-bit) 

## Reproduction Details

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [x] I have included the browser console logs.
- [ ] ~~I have included the Docker container logs.~~

## Logs and Screenshots

**Browser Console Logs:**
```
(index):77 
        
        
       GET https://s.go-mpulse.net/boomerang/B5G2M-TLF9C-X5GW4-VG96U-5SZ7Q net::ERR_CONNECTION_REFUSED
(anonymous) @ (index):77
(anonymous) @ (index):77
(anonymous) @ (index):77
(index):77 
        
        
       GET https://s.go-mpulse.net/boomerang/B5G2M-TLF9C-X5GW4-VG96U-5SZ7Q net::ERR_CONNECTION_REFUSED
t @ (index):77
O._boomrl @ (index):77
t @ (index):77
(anonymous) @ (index):77
load
t @ (index):77
(anonymous) @ (index):77
error
(anonymous) @ (index):77
(anonymous) @ (index):77
(anonymous) @ (index):77
+layout.svelte:88 Backend config: {status: true, name: 'DG AI-Chat (Open WebUI)', version: '0.3.10', default_locale: '', default_models: 'anthropic/claude-3.5-sonnet', …}
+layout.svelte:125 user-count {count: 3}
+layout.svelte:130 usage {models: Array(2)}
+layout.svelte:119 connected
index.ts:64 (178) [{…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, …]
openwebui.url/:1 The resource https://s.go-mpulse.net/boomerang/B5G2M-TLF9C-X5GW4-VG96U-5SZ7Q was preloaded using link preload but not used within a few seconds from the window's load event. Please make sure it has an appropriate `as` value and it is preloaded intentionally.
openwebui.url/:1 The resource https://s.go-mpulse.net/boomerang/B5G2M-TLF9C-X5GW4-VG96U-5SZ7Q was preloaded using link preload but not used within a few seconds from the window's load event. Please make sure it has an appropriate `as` value and it is preloaded intentionally.
```

## Additional Information

- This behavior is not observed with other web applications such as Outlook, Teams, Confluence, or Jira.
- There appears to be no manual override for this wake lock within the browser settings.
- It's unclear whether this is an intentional feature or an unintended bug.
- A potential solution could be to either remove this wake lock or provide a configuration option in the OpenWebUI settings to control this behavior.

**Comment by bold84** [comment]
<img width="1081" alt="Screenshot 2024-08-27 at 03 39 17" src="https://github.com/user-attachments/assets/e4a94114-af01-4354-8d6b-9ab1e4fb9e51">


Same thing here on a Mac.
